src/GitHub.Api/Git/Repository.cs

```diff
@@ -14,20 +14,40 @@ class Repository : IRepository, IEquatable<Repository>
         private ConfigBranch? currentBranch;
         private ConfigRemote? currentRemote;
         private GitStatus currentStatus;
+        private string head;
+        private Dictionary<string, ConfigBranch> localBranches = new Dictionary<string, ConfigBranch>();
+        private IEnumerable<GitLock> locks;
+        private Dictionary<string, Dictionary<string, ConfigBranch>> remoteBranches = new Dictionary<string, Dictionary<string, ConfigBranch>>();
+        private Dictionary<string, ConfigRemote> remotes;
```

Comment:
The purpose of this pull request is here.
These fields are part of the state of the Repository and I am moving them away from `RepositoryManager`.